File: Raven.Client.Java/src/main/java/net/ravendb/client/document/batches/ILazyOperation.java
package net.ravendb.client.document.batches;

import net.ravendb.abstractions.data.GetRequest;
import net.ravendb.abstractions.data.GetResponse;

public interface ILazyOperation {
  public GetRequest createRequest();
  public Object getResult();
  public boolean isRequiresRetry();
  public void handleResponse(GetResponse response);
  public AutoCloseable enterContext();

}

File: src/Raven.Client/Document/Batches/ILazyOperation.cs
using Raven.Client.Data;
using Raven.Client.Data.Queries;

namespace Raven.Client.Document.Batches
{
    public interface ILazyOperation
    {
        GetRequest CreateRequest();
        object Result { get; }
        QueryResult QueryResult { get; }
        bool RequiresRetry { get; }
        void HandleResponse(GetResponse response);
    }
}
